Userland/Libraries/LibArchive/Zip.cpp

```diff
@@ -88,12 +88,12 @@ bool Zip::for_each_member(Function<IterationDecision(ZipMember const&)> callback
         char null_terminated_name[central_directory_record.name_length + 1];
         memcpy(null_terminated_name, central_directory_record.name, central_directory_record.name_length);
         null_terminated_name[central_directory_record.name_length] = 0;
-        member.name = DeprecatedString { null_terminated_name };
+        member.name = TRY(String::from_utf8({ null_terminated_name, strlen(null_terminated_name) }));
```

Comment:
This was already there, but copying and null-terminating the data here just to call `strlen` doesn't seem to be required either, we could just construct the string via `{ central_directory_record.name, central_directory_record.name_length }`.